dstep/translator/MacroDefinition.d

```diff
@@ -436,24 +441,68 @@ class StringConcat : Expression
     }
 }
 
-class StringsSequence : Expression
+class TokenConcat : Expression
 {
-    StringLiteral[] literals;
+    Expression[] subexprs;
+
+    this (Expression[] subexprs)
+    {
+        this.subexprs = subexprs;
+    }
+
+    override string translate(Context context, Set!string params, ref Set!string imports)
+    {
+        import std.algorithm.iteration : map;
+        import std.array : join;
+        import std.format : format;
+
+        string fun(Expression subexpr)
```

Comment:
Ideally you should try to come up with a name that actually describes that the function does, just as if it was a non-nested function.